File: C++Algorithm/DataStructures/Deque/10866/10866.cpp
#include <deque>
#include <iostream>
#include <string>

using namespace std;

int main(int argc, char** argv)
{

    int N;
    cin >> N;

    deque<int> dq;

    for (int i = 0; i < N; i++) {
        string cmd;

        cin >> cmd;

        if (cmd == "push_front") {
            int input;
            cin >> input;
            dq.push_front(input);

        } else if (cmd == "push_back") {
            int input;
            cin >> input;
            dq.push_back(input);

        } else if (cmd == "pop_front") {

            if (dq.empty()) {
                cout << -1 << endl;
            } else {
                cout << dq.front() << endl;
                dq.pop_front();
            }

        } else if (cmd == "pop_back") {

            if (dq.empty()) {
                cout << -1 << endl;
            } else {
                cout << dq.back() << endl;
                dq.pop_back();
            }

        } else if (cmd == "size") {
            cout << dq.size() << endl;

        } else if (cmd == "empty") {
            if (dq.empty()) {
                cout << 1 << endl;
            } else {
                cout << 0 << endl;
            }

        } else if (cmd == "front") {
            if (dq.empty())
                cout << -1 << endl;
            else
                cout << dq.front() << endl;

        } else if (cmd == "back") {
            if (dq.empty())
                cout << -1 << endl;
            else
                cout << dq.back() << endl;
        }
    }

    return 0;
}

File: Algorithm/Baekjun/Python/DataStructures/Deque/10866/10866.py
#!/usr/bin/env python
# coding: utf-8

# In[11]:


from collections import deque
import sys

N = input();

dq = deque()

for i in range(N) :
    Order = list(input().split())
    
    if Order[0] == "push_front" :
        dq.appendleftl(Order[1])
        
    elif Order[0] == "push_back" :
        dq.append(Order[1])
    
    elif Order[0] == "pop_front" :
        if len(dq) == 0 :
            print(-1)
        else :
            print(dq.popleft())
            
    elif Order[0] == "push_back" :
        if len(dq) == 0 :
            print(-1)
        else :
            print(dq.pop())
        
        
    elif Order[0] == "size" :
        print(len(dq))
        
        
        
    elif Order[0] == "empty" :
        if len(dq) == 0 :
            print(1)
        else : 
            print(0)
        
        
    elif Order[0] == "front" :
        
        if len(dq) == 0 :
            print(-1)
        else : 
            print(dq[0])
        
        
    elif Order[0] == "back" :
        if len(dq) == 0 :
            print(-1)
        else : 
            print(dq[len(dq) -1])
        
    


# In[ ]:





# In[ ]:




